dangerzone/conversion/doc_to_pixels.py

```diff
@@ -326,16 +346,26 @@ def pdftoppm_progress_callback(line: bytes) -> None:
             timeout=timeout,
         )
 
-        self.update_progress("Converted document to pixels")
-
-        # Move converted files into /tmp/dangerzone
-        for filename in (
+        final_files = (
             glob.glob("/tmp/page-*.rgb")
             + glob.glob("/tmp/page-*.width")
             + glob.glob("/tmp/page-*.height")
-        ):
+        )
+
+        # XXX: Sanity check to avoid situations like #560.
+        if len(final_files) != 3 * num_pages:
+            raise errors.UnexpectedConversionError(
+                "The final number of pages does not correspond to the page count of the"
+                " original document"
```

Comment:
I don't think this will reach the client. We only pass the error code. So we may want a dedicated error message for this edge-case.